app/src/main/java/com/example/shorebuddy/data/lakes/Lake.java

```diff
@@ -1,13 +1,60 @@
 package com.example.shorebuddy.data.lakes;
 
+
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.PrimaryKey;
+import androidx.room.TypeConverters;
+
+import com.example.shorebuddy.utilities.Converters;
 import com.google.android.gms.maps.model.LatLng;
 
+import org.jetbrains.annotations.NotNull;
+
 // TODO: Implement as a Room model
+@Entity(tableName = "lake_table")
 public class Lake {
+
+    @PrimaryKey
+    @NotNull
     public final String name;
-    public LatLng location = new LatLng(0, 0);
 
-    public Lake(String n) {
-        name = n;
+    @ColumnInfo(name = "Coordinates")
+    @TypeConverters(Converters.class)
+    public LatLng location = new LatLng(0,0);
```

Comment:
We should save location data to our database in a uniform way. Our other model classes, Weather and Solunar use private latitude and longitude fields that can be accessed via getters and setters that use the LatLng type. Neither way is at all wrong but we should standardize.

Reply:
I agree with the standardizing, the reason why there is a LatLng object in the Lake class is because in the MainViewModel.java lines 43 and 50, use a currentLake.location for the getWeatherData and the getSonarData methods. I can remove the LatLng object from the Lake class to have latitude and longitude fields but I need to change the parameters for both of methods that I mentioned above. What would you recommend?

Reply:
Feel free to make any changes in MainViewModel that you need to. I would recommend that the lake class store it's data in two separate fields but provides access to them with getter and setter methods that return the LatLng type. This way room can store them separately but other classes can get access to the LatLng type they need